src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -107,6 +107,22 @@ export const walletSupportsChain = ({
   }
 }
 
+export const walletSupportsChain = ({
+  chainId,
+  chainAccountIds,
+  wallet,
+  isSnapInstalled,
+}: WalletSupportsChainArgs): boolean | null => {
```

Comment:
This function signature doesn't look right, it should never return `null`.

```suggestion
}: WalletSupportsChainArgs): boolean => {
```

Reply:
nice fb777ad518fd15a00ea25df31eb245f0e67e8415